### PR TITLE
Add EOS RAM adapter

### PR DESCRIPTION
**Pull request by DenisCarriere**
##### Name (to be shown on DefiLlama):

EOS RAM

##### Twitter Link:

https://twitter.com/EOSNetworkFDN

##### List of audit links if any:
- N/A

##### Website Link:

- https://docs.eosnetwork.com/docs/latest/core-concepts/resources

##### Logo (High resolution, will be shown with rounded borders):

Same as EOS REX:
- https://icons.llamao.fi/icons/protocols/eos-rex

##### Current TVL:

- $7.49M USD (8,755,388.2038 EOS)

##### Treasury Addresses (if the protocol has treasury)

- N/A

##### Chain:
- EOS

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)

- `eos`
- https://www.coingecko.com/en/coins/eos

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)

- 1765
- https://coinmarketcap.com/currencies/eos/

##### Short Description (to be shown on DefiLlama):

RAM can be purchased and sold by users on the EOS blockchain. The price of RAM is determined by a Bancor algorithm that is implemented in the system contract. The price of RAM is determined by the amount of free RAM available. The less free RAM available, the more expensive it is to buy RAM.

##### Token address and ticker if any:

- N/A

##### Category (full list at https://defillama.com/categories) *Please choose only one:

- Dexes (uses Bancor algorithm)

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):

- N/A
##### forkedFrom (Does your project originate from another project):
- N/A

##### methodology (what is being counted as tvl, how is tvl being calculated):

- EOS RAM TVL is achieved by querying token balances from [`eosio.ram`] system account.
- https://bloks.io/account/eosio.ram

##### Github org/user (Optional, if your code is open source, we can track activity):

- https://github.com/eosnetworkfoundation/eos-system-contracts
- https://github.com/eosnetworkfoundation/eos-system-contracts/blob/main/contracts/eosio.system/src/exchange_state.cpp

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/ram exports TVL: 
        
 
 
    thundercore               314.11 k
    thundercore-borrowed      116.63 k
    borrowed                  116.63 k
    
    total                    314.11 k 
    
    Error writing cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:257:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:238:11
        at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    } BlockCache.json
    Error writing cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/currentChainBlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:257:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:238:11
        at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/currentChainBlockCache.json'
    } currentChainBlockCache.json

File: projects/ram/index.js
Comment (g1nt0ki) [inline projects/ram/index.js:1]: Can you rename the folder as eos-ram? I think it is clashing with ram.js atm, you can ignore the file error

Comment (DenisCarriere) [reply projects/ram/index.js:1]: Ah yes, ok, will rename 👍

**Comment by llamatester** [comment]
Error while running adapter at undefined: 
        
 
  \n Please revert changes to package.json / package-lock.json

**Comment by DenisCarriere** [comment]
@g1nt0ki renamed to `eos-ram` folder

Not sure why `Test_Change` causes some issue, there is no change to `package.json`

**Comment by g1nt0ki** [comment]
I introduced the bug, fixed it now

**Comment by llamatester** [comment]
Error while running adapter at undefined: 
        
 
  \n Please revert changes to package.json / package-lock.json

**Comment by DenisCarriere** [comment]
@g1nt0ki noticed EOS RAM is as "liquid staking" and counted as "Double Count"

This isn't accurate, EOS RAM is the native system DEX to trade RAM/EOS.

This should be considered: "Dexes"

image

However, there's a new product launched by Defibox called bRAM, which is "RAM wrapper"

This one could be considered "double count" since it wraps the native System RAM

https://defibox.io/ram/

These are two different protocols

**Pull request merged**